projects/cal/src/lib/components/kc-cal-day/kc-cal-day.component.ts

```diff
@@ -97,13 +101,41 @@ export class KcCalDayComponent<T extends KcCalSelector = KcCalSelector> implemen
     return this._isDateEqual(this.day, this._removeTime(new Date()));
   }
 
+  @HostBinding('class.kc-cal-day--hovered')
+  get hovered() {
+    if (this.selected || this.other || this.disabled || !this._selector.hovered) return;
+
+    const { from, to, hovered } = this._selector;
+
+    // when ranges are not set, hover based on internal state
+    if (!from && !to) return this._isHovered;
+
+    // when 'to' is not set, hover when is between from & hovered dates
+    if (!!from && !to)
+      return this._isDateBetween(this.day, from, hovered) || this._isDateBetween(this.day, hovered, from);
+
+    // hover when is between either from/to & hovered dates
+    return (
+      (!!from && this._isDateBetween(this.day, from, hovered)) || (!!to && this._isDateBetween(this.day, hovered, to))
+    );
```

Comment:
add tests for all cases.

A case that I don't think will work:
The user first sets the to date and then selects a form that is greater than to

![Peek 2023-12-28 11-28](https://github.com/k5cjs/k5cjs/assets/19764133/3a340736-ac47-4f46-a46c-93808292b801)